lib/cdo/secrets_config.rb

```diff
@@ -31,7 +31,7 @@ def self.load(filename)
       results
     end
 
-    def freeze
+    def freeze_config
```

Comment:
Mostly because we don't want to actually freeze the `Config` object itself or to in any way take advantage of existing `freeze` functionality; we want to implement something analogous but distinct to native `freeze`. If the way we implemented this was to instead call `@table.freeze` and rely on the table being immutable for our table mutation methods to fail, then it'd make sense to me to handle this by overloading `freeze` on the `Config` object, but because we do something much more bespoke it feels right that it's all happening in a custom method.

Reply:
Maybe name the method `ice` instead? That implies the values will sometimes be thawed :)